DhsTranslator/src/DhsAdapterSim.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * DhsAdapterSim.cpp
+ *
+ *  Created on: Feb 18, 2015
+ *      Author: jluhrs
+ */
+
+#include "defines.h"
+#include "DhsAdapterSim.h"
+#include <ctime>
+#include <locale>
+#include <sstream>
+#include <iomanip>
+#include <algorithm>
+#include <iterator>
+
+using namespace std;
+
+DhsAdapterSim::DhsAdapterSim(string &myName, string &serverHost,
+        string &serverName, axutil_log_t * log) {
+    this->myName = myName;
+    this->serverHost = serverHost;
+    this->serverName = serverName;
+    this->nextId = 1;
+    this->log = log;
+
+    pthread_mutex_init(&lock, NULL);
+}
+
+DhsAdapterSim::~DhsAdapterSim() {
+    pthread_mutex_destroy(&lock);
+}
+
+DHS_STATUS DhsAdapterSim::createImage(ImageId &id) {
+    time_t t;
+    time(&t);
+    struct tm *local = localtime(&t);
+    int latchedId;
+
+    pthread_mutex_lock(&lock);
+    {
+        latchedId = nextId;
+        if (++nextId > MAX_IMAGE_ID) {
```

Comment:
Should this be `>=` or do you want to allow `10000` as an ID? If I remember correctly, `++nextId` preincrements and then returns the incremented value, so it is possible that `nextId` could be `10000` and then will not trigger the `if` condition.